src/Taboo/TabooRoom.py

```diff
@@ -127,6 +129,37 @@ def __processKickoff(self, qmsg):
 
         return self.turnMgr.processKickoff(qmsg)
 
+    def __processReady(self, qmsg):
+        """
+        ["READY"]
+        """
+        ws = qmsg.initiatorWs
+
+        if len(qmsg.jmsg) != 1:
+            self.txQueue.put_nowait(ClientTxMsg(["READY-BAD", "Invalid message length"],
+                                                {ws}, initiatorWs=ws))
+            return True
+
+        # No need to do this, right?
+        #if self.state != GameState.WAITING_TO_START:
+        #    self.txQueue.put_nowait(ClientTxMsg(["READY-BAD", "Game already started/ended"],
+        #                            {ws}, initiatorWs=ws))
+        #    return True
```

Comment:
Ok will do in next pull request

Reply:
We need to handle two cases of late-joinees:

Case 1: Player joins after the game starts

Case 2: Player re-joins before or after the game starts
 
My proposal is this: 

if received READY for plyr:
    if plyr.ready:
         send ['READY-BAD', 'already ready'] to initiating ws
   else: 
        plyr.ready = True

Definition: connected player = player with at least one connected ws
Definition: active player = a connected player that is ready

Game start is triggered only once: when all currently connected players are active and there are at least two active players in each team
e.g., a READY for a late-joinee or a re-joinee does not trigger startNewGame(). It will just make it an active player (and thus will get a play-turn)
  
A disconnected player's ready status is reset. i.e.,
if plyr lost its last ws: 
   plyr.ready = False

This means, for Case 2, a player that re-joins later is only admitted to the GAME as active player when it explicitly sends READY after the (re)JOIN.

Comments?

Reply:
Notifying READY-BAD makes sense (we should capture this in README.md too)

Late player joinees (not late connections to existing players) should immediately be flagged as ready (just for consistency?).

I don't think we should reset the ready state though. We can already track live players (playerConns.count() > 0). Let's not overload the meaning of ready by also tracking number of clients.

Reply:
Agree. Will do.